### DISCUSSION TITLE
Should highway=bus_stop show local_ref in the quest title?

### DISCUSSION THREAD

**Discussion by kmpoppe** [streetcomplete/StreetComplete#3800] [Q&A] [answered]
This one's especially for @peternewman as someone mapping a lot in London, but as a frequent visitor, I've got my own opinion on this (and you might have guessed it from the title already 😉).

In areas where there are lots and lots of bus stops in close proximity it could be beneficial to not only show the `name` but also the `local_ref` in brackets behind the name to distinguish the quest objects from one another.

There are 170k uses of that key, so the effort of just changing the strings around and adding the value in the code will not be made for just a handful of cases - and also is not a big deal.

Opinions?

Have a great day, everyone!

Kai

**Comment by andrewharvey** [comment]
Agreed.

**Comment by matkoniecz** [comment]
I see that it is an imported tag. Is it really not considered as part of bus stop name?

Asking as in Poland it would be tagged as part of `name` tag and NAPTAN import had some peculiarities and mistakes which should not be encouraged. It may be one of them.

**Reply by andrewharvey** [reply]
According to https://lists.openstreetmap.org/pipermail/talk-au/2022-February/015664.html

> local_ref has been used it at least the UK and Sweden to tag the stand 
number. It doesn't (yet) render on the OSM map main layer but does on 
the transport layers.    /Mike

**Reply by kmpoppe** [reply]
One could always use a staggered approach with

local_ref/loc_ref (local)
reg_ref (regional)
nat_ref (national)
int_ref (international)
ref (all-purpose)

being used in that order to show a distinction - because sadly, where I live the different bus/tram stops are all called "Hauptbahnhof" (Main station) and only `ref` (and `description`) having the distinctive values...

**Reply by matkoniecz** [reply]
It was definitely **used** (mostly in import). The question is it sole recommended method in places where it was used? Is it valid to use it when it is already in `name`?

I worry that someone would add it in say Poland where `name=Dworzec kolejowy`, `name=Dworzec kolejowy 2` would be a valid. So someone would add technically valid `local_ref=1` and StreetComplete would display it as "Dworzec kolejowy 1 1" or "Dworzec kolejowy 1 (1)"

**Reply by kmpoppe** [reply]
As the quest is not asking for the ref and merely displaying it, what you describe would come down to a tagging mistake against the locally accepted rules, right?

Then again, the logic for displaying the ref could check if the name starts or ends with the same character-sequence as the ref and then not add the ref to the quest title.

**Reply by kmpoppe** [reply]
> I see that it is an imported tag. Is it really not considered as part of bus stop name?

To answer your initial question, at least in Germany (and from what I know from the UK), the ref is not considered part of the name of the stop and we always split them from the name.

**Reply by andrewharvey** [reply]
>  StreetComplete would display it as "Dworzec kolejowy 1 1" or "Dworzec kolejowy 1 (1)"

It's just StreetComplete, it's not a proper downstream application, so we shouldn't be too worried about it looking a bit wrong. We should instead prioritise showing more detail to help mappers make less errors to give more accurate data.

**Reply by peternewman** [reply]
It's a discrete piece of information @matkoniecz therefore it should be, and has been captured separately. Think of it the same as platforms at a railway station.

See:
https://commons.wikimedia.org/wiki/File:London_Bus_route_15.jpg
I was going to write out all the keys, but realised I don't need to, as it's just this node:
https://www.openstreetmap.org/node/469761460

The name tells me where I am/that I'm in the right area, the local_ref tells me that I'm at the right stop/side of the road/not round the corner. At some busy/important places there can be two or more bus stops on the same side of the road next to each other, buses going NW say might stop at the front one, and NE at the back one (I don't know if it's that simple), then SW and SE on the other side, so you need to get the right one!

I'd agree if the 2 or 55 here are the "local_ref" they're definitely in the name (although I'd be confused by the 55 unless all stops on the route/road are numbered consecutively):
https://www.alamy.com/stock-photo-a-bus-passenger-viewing-a-bus-time-table-at-a-bus-stop-in-warsaw-poland-133954166.html
https://www.rgbstock.com/photo/mhixt3e/Bus+stop

**Reply by peternewman** [reply]
Plus in terms of the original question, I think this would have allowed me to answer a few questions rather than leaving notes, especially when there is a mismatch in stop numbers between on the map and in real life.

**Comment by peternewman** [comment]
I never knew you cared @kmpoppe :laughing: !

**Reply by kmpoppe** [reply]
Giving you a smiley for the mention, yet I'm unsure what you mean exactly 😉

**Reply by peternewman** [reply]
Just the fact you've noticed I map in London a lot!

**Reply by kmpoppe** [reply]
Easy 😉 -> sees disused:amenity issue by you -> goes to area -> randomly taps Bus Stop Bin Quest -> thinks about discussed problem -> goes back to Node on OSM.org and sees you last edited it -> looks at your recent edits in OSM profile ... 6 easy steps to guessing where your hotspot is :)

**Comment by kmpoppe** [comment]
This is what I've come up so far (working with AddBinStatusOnBusStop).

**Screenshots:**

London, UK, Warren Street Station/Tottenham Court Road:
https://www.openstreetmap.org/node/469760369
`local_ref` and `ref` exist, `local_ref` takes presendece.
![grafik](https://user-images.githubusercontent.com/24451207/155350942-dcb0be5e-a5c9-432a-abc5-894e68766318.png)

Osnabrück, Germany, Weißenburger Straße
https://www.openstreetmap.org/node/535436703
Only `ref` is set.
![grafik](https://user-images.githubusercontent.com/24451207/155351422-9a6682cd-3684-4910-8c86-dda10705d881.png)

Warsaw, Poland, Krucza
https://www.openstreetmap.org/node/334558429
`ref` is set and the last two characters equal the last two characters of the `name`. Special treatment for Poland.
![grafik](https://user-images.githubusercontent.com/24451207/155351839-a4fad38c-6bf7-4339-a0af-58835bb996a5.png)

**Code changes:**

e5bbc5c

What y'all think?

**Reply by peternewman** [reply]
> Warsaw, Poland, Krucza https://www.openstreetmap.org/node/334558429 `ref` is set and the last two characters equal the last two characters of the `name`. Special treatment for Poland.

This smells a bit funny to me, feels like it may cause some odd behaviour somewhere. Given Poland's ref is longer, I don't really see the harm in showing the whole thing, it's not like repeating the V at the end of the London one.

> **Code changes:**
> 
> [e5bbc5c](https://github.com/streetcomplete/StreetComplete/commit/e5bbc5cf8fd82bde3bcd4395475725e9e6c2dd1b)

What happens if you try a stop with no name set at all, i.e. null rather than empty, and a ref, won't you get the ref twice?

> What y'all think?

I wonder if the ref bit wants the word ref (so "(ref V)") to give it a little context, it's going to seem a bit random otherwise, especially if they aren't as clearly marked as the London stops.

**Reply by FloEdelmann** [reply]
> I wonder if the ref bit wants the word ref

That's how it is for fire hydrants:

https://github.com/streetcomplete/StreetComplete/blob/1c5b7b4b964680646d8fd5483cd598a4a17415ac/app/src/main/res/values/strings.xml#L1148

**Reply by matkoniecz** [reply]
oh dear, this specific ref code has likely a different issues (probably not actually used by passengers at all to identify anything - maybe tagging can be improved here, I opened https://forum.openstreetmap.org/viewtopic.php?pid=856364#p856364 )

**Reply by peternewman** [reply]
I see #3870 has gone in, could we look at local_ref again now that's merged?

I know @westnordost didn't want specific tweaks:
https://github.com/streetcomplete/StreetComplete/pull/3870#discussion_r825449814

But maybe just taking local_ref over ref if both are set (and showing ref more often) would cover it (although you'd need to show local_ref/ref more too?

The relevant code is:
https://github.com/streetcomplete/StreetComplete/blob/fef6877852d6a19a7b85e6f3ed3b09ea7c6538ec/app/src/main/java/de/westnordost/streetcomplete/quests/NameAndLocationLabel.kt#L63-L74

The inevitable example that breaks it (although they are further apart than London ones):
https://www.openstreetmap.org/node/502532149
https://www.openstreetmap.org/node/502532146

**Reply by westnordost** [reply]
Looking at the wiki page for `local_ref`, this seems to make sense. Care to create a PR?

**Reply by peternewman** [reply]
Done in #3903 . I suspect this won't be quite right everywhere, but hopefully not too bad in the odd situations like @matkoniecz mentioned.

**Comment by kmpoppe** [comment, marked as answer]
Solved in #3870 and #3903 by @peternewman
